Add New Category screen

EL-MANCY opened the pull request:
![image](https://github.com/user-attachments/assets/192cadf4-ce21-4094-94f9-6984fd8fa08f)
![image](https://github.com/user-attachments/assets/2ddd9c9c-f7da-49f9-a8d1-1d0cb68e39a0)
![image](https://github.com/user-attachments/assets/52edf516-f935-4832-b36d-772321892545)

# **Implement "Add New Category" Bottom Sheet**

## **Description**

This pull request introduces the "Add New Category" feature, presented within a modal bottom sheet. It includes a fully functional form with an interactive image uploader and a text field for the category title.

### **Key Changes**

**1. `UploadBox` Composable**

* Created a new, reusable `UploadBox` component with a dashed border style to serve as a clear drop zone.
* Integrated an image picker that launches the device gallery using `rememberLauncherForActivityResult`.
* The component now handles two distinct states:
    * **Initial State:** An "Upload" prompt.
    * **Selected State:** Displays the user-selected image (cropped to fit) with an "edit" icon overlay, allowing the user to select a different image.

**2. `AddNewCategory` Screen**

* Built the complete UI for the "Add New Category" bottom sheet.
* The `TudeeTextField` for the "Category title" is now stateful, enabling users to input and modify text.
* Successfully integrated the new `UploadBox` component into the screen layout.

### **Dependencies**

* Added the **Coil** library (`io.coil-kt:coil-compose`) to asynchronously load and display the selected image from its URI, ensuring a smooth and responsive user experience.

EL-MANCY committed:
add coil dependency
add AddNewCategorey.kt screen design
add UploadBox.kt as a component

diyar214 reviewed (changes requested):
1- Check the color of `Dashed Border` to match the design.
2- Check the color of the `menu_circle` inside the textField.
3- The `Add Button` should be disabled, then it will be enabled when the user fills the category information.

Other than that you did great job 

<img width="809" alt="Figma Screenshot" src="https://github.com/user-attachments/assets/5cd196b5-c8e8-4cb8-b480-f9e97d1e8d14" />
<img width="900" alt="PR Screenshot" src="https://github.com/user-attachments/assets/0e066033-4070-40ec-858a-15f13df3f1ed" />

haideraljawasim reviewed (changes requested):
every thing, looks Great,  except  for what mentioned in the comments , last note the PR should be on the branch of the screen not on the dev since this part of category screen

EL-MANCY committed:
add a drop shadow to the buttons container
remove fillMaxHeight and wrapped the content with right dimensions

EL-MANCY committed:
-fix padding of the feature/addNewCategoreyBottomSheet

-fix the stroke of UploadBox.kt

EL-MANCY committed:
process the selected image to bitmap

diyar214 reviewed (dismissed):
LGTM!

AimanYosofiUlt reviewed (changes requested):
Good Job, Just please check my leatest comments

EL-MANCY committed:
-add HelperFunctions.kt for the processImage
-add some string resources
-extract multiple parts of the composables to small functions

EL-MANCY committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into feature/addNewCategoreyBottomSheet

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/res/values/strings.xml

EL-MANCY committed:
hoist multiple states
